### PR TITLE
Add new module to support Jakarta EE for Hibernate 5.5

### PR DESCRIPTION
**Pull request by jon5477**
I am in the process of migrating from Java 8 to 11 and unfortunately, this Jackson module does not play nice with the Jakarta-based Hibernate 5.5 releases. I have done the following changes:
- Add module-info using Moditect
- Shade package names from `javax.persistence` to `jakarta.persistence` as needed

This is my first contribution so please let me know if I missed anything, happy to make changes to get this merged in.

### CODE REVIEW

**Comment by cowtowncoder** [comment]
Hi @jon5477! First of all, thank you for your contribution. Hibernate module does indeed miss JPMS support, unlike more other (better maintained) modules.

One quick thing: since this will change behavior, it cannot go in 2.12 or earlier but needs to go in 2.13.0 (if it can make it in time). So this should be rebased against `master` (since Hibernate module is not scheduled to be continued past 2.x `master` is for latest 2.x, not 3.0).

Aside from that, I have some questions about need for shading -- I know it may be necessary, but over time there's been a trend wherein most cases of shading have led to various issues users report.

@GedMarc WDYT?

**Comment by cowtowncoder** [comment]
Oh, on shading: I suppose this tries to solve the dreaded JAXB vs Jakarta. While understandable, that is a big big problem. :(

Unfortunately, doing that will definitely break all preceding 5.x usage. I don't think that is something we can do: existing users' use cases should not be broken by new version without good reason.

Given this, it might be necessary to instead create another sub-module: if the Jakarta change occurred with 5.5 (link to some announcement would be great!), maybe there is need for Jakarta5_5 module. I don't necessarily mind code duplication if it is necessary: in this case it may well be. So it'd be a fully copy of existing `hibernate5` sub-project, with suitable renaming.

**Comment by jon5477** [comment]
Hey, glad to hear your feedback. I am fine with anything as long as I can get this change in (I need it for serializing Hibernate mapped entities into JSON). I definitely want to maintain compatibility and not break anyone's existing code. The documentation for this is located at [Hibernate 5.5 Release](http://hibernate.org/orm/releases/5.5/). Hibernate now allows people to switch between the old `javax.persistence` 2.2 API and the new `jakarta.persistence` 3.0 API. They did this by adding the "jakarta" suffix to all their dependencies so you have `hibernate-core` and `hibernate-core-jakarta` respectively. I think the best course of action for this would be to create an entirely new module for Hibernate 5.5 and keep the existing Hibernate 5.x for legacy users. I will make those changes as needed.

**Comment by cowtowncoder** [comment]
@jon5477 That sounds good -- in addition to this particular problem, I think Hibernate 5.x "minor" versions are mutually incompatible in surprising ways (at least 5.0 / 5.1 / 5.2 had some oddities).
I trust your expertise on getting a new one added and hope we can get this done relatively quickly: will try to help here.

Note that with a new module it should be fine to directly depend on `hibernate-core-jakarta` directly and avoid shade re-packaging? (as there's no backwards-compatibility to be concerned about).
Jackson 2.13 has a few other modules with similar dual-setup; duplication is unfortunate but seems like the "least worst" option.

**Comment by jon5477** [comment]
Alright, I made several changes/updates to this and it should be ready for review.

Some important points:
- I maintained the original hibernate5 module - this will continue to work for Hibernate 5.5 against JPA 2.2
- I added a new hibernate5-jakarta module - this will work for Hibernate 5.5+ against the Jakarta Persistence API 3.0
- Users can either switch between the original hibernate5 module or hibernate5-jakarta module depending on their use-case

Please let me know if there are additional points I missed.

**Comment by cowtowncoder** [comment]
@jon5477 Thank you! I hope to review this soon.

File: README.md [hunk 1]
```diff
@@ -41,6 +41,17 @@ or "jackson-datatype-hibernate5" for Hibernate 5.x
...
 ```
 
+if you plan to use Hibernate 5.5 with the Jakarta Persistence API 3.0;
+you will need the jakarta suffixed dependency for Hibernate 5.5:
+
+```xml
+<dependency>
+    <groupId>com.fasterxml.jackson.datatype</groupId>
+    <artifactId>jackson-datatype-hibernate5-jakarta</artifactId>
+    <version>2.16.0</version>
```
Comment (cowtowncoder) [inline README.md:51]: 2.13.0

**Comment by cowtowncoder** [comment]
Ok great; looks good at high level. Just one (?) thing I need (UNLESS I already asked for some other contribution), CLA. It's here:

https://github.com/FasterXML/jackson/blob/master/contributor-agreement.pdf

and the usual way is to print, fill & sign, scan/photo, email to `info` at fasterxml dot com.

Once I get it, I can merge the PR and we should be good to go.
(I do still hope to review it a bit more, but aside from that).

File: README.md (hunk 1 above)
Comment (jon5477) [reply README.md]: Done.

**Comment by cowtowncoder** [comment]
CLA received, ready for final review & merge!

**Comment by cowtowncoder** [comment]
Thank you @jon5477 for this contribution: merged, will be included in 2.13.0.
I made some minor naming changes and build succeeds. But it would be great if you could verify that 2.13.0-SNAPSHOT from Sonatype OSS repo works: I `mvn deploy`ed a new version before cut the release.

**Comment by jon5477** [comment]
@cowtowncoder Hey, sorry for the late follow-up. I tested it against the latest snapshot (2.13.0-SNAPSHOT) on Sonatype OSS and it is confirmed working for Hibernate 5.5 Jakarta. I was able to serialize my Hibernate entities to JSON without any issue.

**Comment by cowtowncoder** [comment]
Ok good! I just released 2.13.0 so that should be verifiable now too.

**Pull request merged**